File: project/src/main/java/br/com/project/service/EditProfileService.java
package br.com.project.service;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

import br.com.project.models.User;
import br.com.project.repositories.UserRepository;

import javax.transaction.Transactional;

@Service
@Transactional
public class EditProfileService {
	
	@Autowired
	UserRepository userRepository;
	
	public HttpStatus editName(String username, String newName) {
		Optional<User> userOptional = userRepository.findByUsername(username);
		if(username != null) {
			User user = userOptional.get();
			user.setName(newName);
			userRepository.save(user);
			return HttpStatus.ACCEPTED;
		}
		return HttpStatus.BAD_REQUEST;
	}
	
	public HttpStatus editPassword(String username, String newPassword) {
		Optional<User> userOptional = userRepository.findByUsername(username);
		if(username != null) {
			User user = userOptional.get();
			user.setPassword(newPassword);
			BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
			String passwordEncoder = encoder.encode(user.getPassword());
			user.setPassword(passwordEncoder);
			userRepository.save(user);
			return HttpStatus.ACCEPTED;
		}
			return HttpStatus.BAD_REQUEST;
	}
}

File: src/main/kotlin/br/com/project/projetoIntegrador/service/EditProfileService.kt
package br.com.project.projetoIntegrador.service

import br.com.project.projetoIntegrador.dto.EditPasswordDTO
import br.com.project.projetoIntegrador.exceptions.InvalidInputException
import br.com.project.projetoIntegrador.repositories.UserRepository
import br.com.project.projetoIntegrador.utils.InputUtils
import jakarta.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import java.util.*

@Service
@Transactional
class EditProfileService @Autowired constructor(
        private val userRepository: UserRepository,
        private val passwordEncoder: PasswordEncoder
    ) {

    fun editName(username: String, newName: String) {
        val user = userRepository.findByUsername(username).get()
        user.name = newName
        userRepository.save(user)
    }

    fun editPassword(username: String, editPasswordDTO: EditPasswordDTO) {
        val user = userRepository.findByUsername(username).get()
        if(user.password == editPasswordDTO.actualPassword) {
            if(InputUtils.validatePassword(editPasswordDTO.newPassword)) {
                user.password = passwordEncoder.encode(editPasswordDTO.newPassword)
                userRepository.save(user)
            } else throw InvalidInputException("A senha deve conter de 8 a 20 caracteres, pelo menos um caractere em maiúsculo e um caractere especial")
        } else throw InvalidInputException("Senha atual inválida!")
    }
}